cassandra/cluster.py

```diff
@@ -2654,6 +2675,12 @@ def has_more_pages(self):
         """
         return self._paging_state is not None
 
+    @property
+    def custom_payload(self):
+        if not self._event.is_set():
+            raise Exception("custom_payload cannot be retrieved before ResponseFuture is finalized")
```

Comment:
Should this wait for the future to be set instead of throwing if not set? I shouldn't be that much more overhead. Maybe all I care about is the custom payload. :)